portal-frontend/src/app/features/applications/edit-submission/primary-contact/primary-contact.component.ts

```diff
@@ -217,90 +213,77 @@ export class PrimaryContactComponent extends FilesStepComponent implements OnIni
-        this.selectedLocalGovernment = selectedOwner.type.code === OWNER_TYPE.GOVERNMENT;
-
-        this.isExistingOwner.setValue(!this.selectedThirdPartyAgent);
-      }
 
-      if (this.selectedLocalGovernment) {
-        this.organizationName.setValidators([Validators.required]);
-      } else {
-        this.organizationName.setValidators([]);
-      }
+    if (!parcels || !allOwners) {
+      return;
+    }
 
-      if (selectedOwner && (this.selectedThirdPartyAgent || this.selectedLocalGovernment)) {
-        this.selectedOwnerUuid = selectedOwner.uuid;
-        this.form.patchValue({
-          firstName: selectedOwner.firstName,
-          lastName: selectedOwner.lastName,
-          organizationName: selectedOwner.organizationName,
-          phoneNumber: selectedOwner.phoneNumber,
-          email: selectedOwner.email,
-        });
-      } else if (selectedOwner) {
-        this.onSelectOwner(selectedOwner.uuid);
-      } else {
-        this.firstName.disable();
-        this.lastName.disable();
-        this.organizationName.disable();
-        this.email.disable();
-        this.phoneNumber.disable();
-      }
+    this.hasCrownParcels =
+      !!parcels && parcels.some((parcel) => parcel.ownershipTypeCode === PARCEL_OWNERSHIP_TYPE.CROWN);
+
+    const uniqueParcelOwners = parcels
+      .flatMap((parcel) => parcel.owners)
+      .reduce((map, owner) => {
+        return map.set(owner.uuid, owner);
+      }, new Map<string, ApplicationOwnerDto>());
+    const nonParcelOwners = allOwners.filter((owner) =>
+      [OWNER_TYPE.AGENT, OWNER_TYPE.GOVERNMENT].includes(owner.type.code)
+    );
+    const parcelOwners = [...uniqueParcelOwners.values()];
+    const owners = [...parcelOwners, ...nonParcelOwners];
+
+    this.parcelOwners = parcelOwners;
+    this.owners = owners;
+
+    const selectedOwner = owners.find((owner) => owner.uuid === primaryContactOwnerUuid);
+
+    // onSelectOwner only not called on first load of page
+    if (selectedOwner) {
+      this.onSelectOwner(selectedOwner.uuid);
+    } else if (parcelOwners.length === 1) {
+      this.onSelectOwner(parcelOwners[0].uuid);
+    }
 
-      if (this.isGovernmentUser || this.selectedLocalGovernment) {
-        this.prepareGovernmentOwners();
-      }
+    // Needed for L/FNG user on first load
+    this.selectedLocalGovernment = selectedOwner?.type.code === OWNER_TYPE.GOVERNMENT;
+    if (this.isGovernmentUser && this.selectedLocalGovernment) {
+      this.organizationName.setValidators([Validators.required]);
+      this.prepareGovernmentOwners();
+    } else {
+      this.organizationName.setValidators([]);
+    }
 
-      if (this.showErrors) {
-        this.form.markAllAsTouched();
-      }
-      this.isDirty = false;
-      this.calculateLetterRequired();
+    if (this.showErrors) {
+      this.form.markAllAsTouched();
+      this.ownersList.markAsTouched();
     }
+    this.isDirty = false;
+    this.calculateLetterRequired();
   }
```

Comment:
Have you tested to see if changing the value of `isDirty` here will not affect the form validation states that was set in the if clause above? Same thing for NOI